chain/txValidator.go

```diff
@@ -768,30 +775,36 @@ func (bvs *BlockValidationState) VerifyTransactionWithBlock(txn *transaction.Tra
 		if err != nil {
 			return err
 		}
-		amount = common.Fixed64(npPayload.Amount) - channelBalance
+
+		if common.Fixed64(npPayload.Amount) <= channelBalance {
+			return errors.New("invalid nanopay amount")
+		}
+
+		npAmount = common.Fixed64(npPayload.Amount) - channelBalance
 
 		defer func() {
 			if e == nil {
 				bvs.addChange(func() {
 					bvs.nanoPays[key] = struct{}{}
+					bvs.nanoPayPayloads[txn.Hash()] = npPayload
 				})
 			}
 		}()
 
 	case pb.ISSUE_ASSET_TYPE:
 	}
 
-	if amount > 0 || fee > 0 {
+	if amount+npAmount+fee > 0 {
```

Comment:
npAmount can be 0?

Reply:
Yes, it's only non-zero if tx type is nano pay